cmd/celestia/rpc.go

```diff
@@ -440,8 +441,8 @@ func parseNamespace(param string) (namespace.ID, error) {
 			return nil, fmt.Errorf("error decoding namespace ID: %w", err)
 		}
 	}
-	// if the namespace ID is 8 bytes, add v0 share + namespace prefix and zero pad
-	if len(nID) == 8 {
+	// if the namespace ID is <= 10 bytes, add v0 share + namespace prefix and zero pad
+	if len(nID) <= appns.NamespaceVersionZeroIDSize {
```

Comment:
This check happens in the constructor so it's redundant

Reply:
I assumed returning an error here was correct. Do we want to avoid throwing because we want to support full namespaces besides sub-namespaces(<=10)?

If so, I suggest adding logic that detects if a namespace is full or sub and returns instantiated namespace. 
Something like `share.DecodeNamespace([]byte) Namespace`, would enable anyone to take a namespace of any form and simply use it.

Reply:
Ideally, we would be able to accept all these 3 cases, and something like Decode/Parse would help. Ideally, we would also move the Parse/Decode helper next to app's namespace. How does that sound to you?

Reply:
> Ideally, we would be able to accept all these 3 cases

Are these the cases you're considering:

version | ID | namespace | example
--- | --- | --- | ---
✅ | ✅ | computed based on other params | `celestia rpc blob Submit 0x42690c204d39600fddd3 --namespace-version 0`
❌ (defaults to `0`) | ✅  | computed based on other params | `celestia rpc blob Submit 0x42690c204d39600fddd3`
❌ | ❌ | ✅ | `celestia rpc blob Submit 0x0000000000000000000000000000000000000001010101010101010101`

If so, I think it's confusing that the param in most cases accepts a namespace ID but in the last case accepts a namespace.

> Ideally, we would also move the Parse/Decode helper next to app's namespace

I'm onboard with extracting a helper from celestia-node to celestia-app if that's what is suggested. And/or moving it to a share or namespace package inside this repo. Defer to ya'll on where `parseNamespace` should live.

Reply:
Also we really need to clarify to users if they should be providing a 10 byte namespace ID, a partial namespace ID (e.g. 8 bytes), or a full namespace.

Reply:
passing this to @distractedm1nd as you will have a call today

Reply:
From our sync discussion today:
> `namespace.Parse/Decode([]byte) Namespace` Which accepts:
    * Ver + Padded ID
    * Padded ID
    * ID
    * Maybe Ver + ID

> The text encoding is out of the scope of this function, so parseNamespace will stay, but just use it.